src/main.rs

```diff
@@ -0,0 +1,264 @@
+    let args = Args::parse();
+    info!("Started: {:#}", args);
+
+    let c_info = client_info::ClientInfo::new(&args)?;
+    info!("{:#?}", c_info);
+
+    let rng = &mut thread_rng();
+
+    let max_degree = 1 << 15;
+    let max_config = PuzzleConfig { degree: max_degree };
+    info!("Loading CoinbasePuzzle...");
+    let universal_srs = CoinbasePuzzle::<Testnet3>::setup(max_config).unwrap();
+
+    let degree = (1 << 13) - 1;
+    let config = PuzzleConfig { degree };
+    let coinbase_puzzle = CoinbasePuzzleInst::trim(&universal_srs, config).unwrap();
+    let (epoch_challenge, address, _) = sample_inputs(degree, rng);
+
+    let challenge = Challenge { task_id: "zk_games".to_string(), epoch_challenge: epoch_challenge.clone() };
+
+    let results_file_path = format!("{}_{}", epoch_challenge.clone().epoch_number(), consts::RESULTS_FILE_TEMPLATE);
+    let mut results_file = File::create(results_file_path).unwrap();
+    results_file.write(format!("{:#?}", c_info).as_bytes());
+    results_file.write_all(b"===coinbase_target===");
+    results_file.write_all(&consts::SHARE_TARGET.to_bytes_le().unwrap());
+    results_file.write_all(b"===proof_target===");
+    results_file.write_all(&[0u8]); //
+    results_file.write_all(b"===epoch_challenge===");
+    results_file.write_all(&epoch_challenge.to_bytes_le().unwrap());
+    results_file.write_all(b"===max_degree===");
+    results_file.write_all(&max_degree.to_bytes_le().unwrap());
+    results_file.write_all(b"===degree===");
+    results_file.write_all(&degree.to_bytes_le().unwrap());
+    results_file.write_all(b"===solutions===").unwrap();
+
+    tokio::spawn(do_puzzle(args.clone(), coinbase_puzzle, challenge, address));
+    Ok(())
+}
+
+async fn do_puzzle<N: Network>(
+    args: Args,
+    coinbase_puzzle: CoinbasePuzzle<N>,
+    challenge: Challenge<N>,
+    address: Address<N>,
+) {
+    let threads_num = args.threads_num;
+    let parallel_num = args.parallel_num;
+
+    let mut thread_pools = Vec::new();
+
+    for _ in 0..parallel_num {
+        let rayon_panic_handler = move |err: Box<dyn core::any::Any + Send>| {
+            error!("{:?} - just skip", err);
+        };
+        thread_pools.push(Arc::new(
+            ThreadPoolBuilder::new()
+                .stack_size(consts::THREAD_STACK_SIZE)
+                .num_threads(threads_num as usize)
+                .panic_handler(rayon_panic_handler)
+                .build()
+                .expect("Failed to initialize a thread pool for worker using cpu"),
+        ));
+    }
+
+    let mut provers_state = start_provers(thread_pools.clone(), coinbase_puzzle.clone(), challenge.clone(), address);
+
+    loop {
+        std::thread::sleep(Duration::from_millis(5000));
+
+        let task_id = provers_state.task_id.clone();
+        let elapsed = provers_state.proves_start.read().unwrap().elapsed().as_secs_f64();
+        let iterations = provers_state.proves_count.load(Ordering::SeqCst) as usize;
+        if elapsed >= 60f64 {
```

Comment:
`60f64` can this be constant with an explanation plz

Reply:
y, will do